From 5d64f11e3787a357cd429b77a91c363d73072985 Mon Sep 17 00:00:00 2001
From: MengLi <meng.li@windriver.com>
Date: Sat, 18 Apr 2020 18:24:04 +0800
Subject: [PATCH] Revert "crypto: caam: Fix compilation warnings"

This reverts commit 832585f2aee5bf7ce4cade0dfb45bbd0d67f0e37.

This patch is used to fix build warning in functions
ablkcipher_setkey and xts_ablkcipher_setkey, but the 2 functions
had been removed in v5.2 kernel. When integrate patches from
SDK, This patch is involed by mistake and used to modify code in
2 functions skcipher_setkey and xts_skcipher_setkey, and then
casue caam feature doesn't work.
So, remove this disruptive patch.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/crypto/caam/caamalg.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/crypto/caam/caamalg.c b/drivers/crypto/caam/caamalg.c
index 575d095d4c3c..0319e02fac61 100644
--- a/drivers/crypto/caam/caamalg.c
+++ b/drivers/crypto/caam/caamalg.c
@@ -780,7 +780,7 @@ static int skcipher_setkey(struct crypto_skcipher *skcipher, const u8 *key,
 	}
 
 	ctx->cdata.keylen = keylen;
-	ctx->cdata.key_virt = ctx->key;
+	ctx->cdata.key_virt = key;
 	ctx->cdata.key_inline = true;
 
 	/* skcipher_encrypt shared descriptor */
@@ -835,7 +835,7 @@ static int xts_skcipher_setkey(struct crypto_skcipher *skcipher, const u8 *key,
 	}
 
 	ctx->cdata.keylen = keylen;
-	ctx->cdata.key_virt = ctx->key;
+	ctx->cdata.key_virt = key;
 	ctx->cdata.key_inline = true;
 
 	/* xts_skcipher_encrypt shared descriptor */
-- 
2.17.1

